numpyro/distributions/continuous.py

```diff
@@ -1143,8 +1143,6 @@ def log_prob(self, value):
         log_terms = np.log(two_n_plus_one) - 1.5 * np.log(value) - 0.125 * np.square(two_n_plus_one) / value
         even_terms = np.take(log_terms, all_indices[::2], axis=-1)
         odd_terms = np.take(log_terms, all_indices[1::2], axis=-1)
-        logsumexp_even = logsumexp(even_terms, axis=-1)
-        logsumexp_odd = logsumexp(odd_terms, axis=-1)
-        even_odd_ratio = np.exp(logsumexp_odd - logsumexp_even)
-        even_odd_ratio = np.clip(even_odd_ratio, a_max=1.0 - 1.0e-6)
-        return logsumexp_even + np.log1p(-even_odd_ratio) - 0.5 * np.log(2.0 * np.pi)
+        sum_even = np.exp(logsumexp(even_terms, axis=-1))
+        sum_odd = np.exp(logsumexp(odd_terms, axis=-1))
+        return np.log(sum_even - sum_odd) - 0.5 * np.log(2.0 * np.pi)
```

Comment:
I think this will fail if `sum_even < sum_odd`. Maybe we can clamp the diff by `np.finfo(value).tiny`?

Reply:
this will never happen in the restricted range of the domain. this is one reason for restricting the range